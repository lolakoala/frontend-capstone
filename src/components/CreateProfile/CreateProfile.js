import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Chip, RaisedButton } from 'material-ui';
import { Field, reduxForm } from 'redux-form';
import { TextField } from 'redux-form-material-ui';

const validate = values => {
  const errors = {};
  const requiredFields = ['userName', 'aboutMe'];
  requiredFields.forEach(field => {
    if (!values[field]) {
      errors[field] = 'Required';
    }
  });
  return errors;
};

class CreateProfile extends Component {

  componentWillMount() {
    //get list of challenges from store
    this.props.getAllChallenges();
  }

  uploadPhoto = () => {
    //dunno wtf will happen here
  }

  addChallenge = challenge => {
    // change color of chip
    // add challenge to userChallenges
    this.props.addUserChallenge(challenge);
  }

  renderChip = (challenge, index) => {
    return (
      <Chip
        key={`${challenge}${index}`}
        onClick={() => { this.addChallenge(challenge); }}>
        {challenge}
      </Chip>
    );
  }

  submitProfile = () => {
    // if we make another form, things might get hairy in the form object we're accessing from the store?
    const { submitProfile, currentUser, userChallenges, form } = this.props;
    // userImage key on this object to send image?
    const newUser = Object.assign(
      {},
      currentUser,
      { userChallenges },
      { userName: form['object Object'].values.userName, aboutMe: form['object Object'].values.aboutMe }
    );
    // somehow send image in submitProfile action?
    submitProfile(newUser);
  }

  //logo - from persisted component?

  render() {
    const userKeys = Object.keys(this.props.currentUser);
    if (userKeys.length > 2) {
      return <Redirect to="/editProfile" />;
    }
    return (<div>
      <form>
        {/* if user has photo, display photo and button to change it (or adjust button below to say 'Change Photo') */}
        <RaisedButton label="Upload Photo" onClick={this.uploadPhoto} />
        <Field
          name="userName"
          component={TextField}
          hintText="Username: Must be at least 5 characters."
          type="text"
        />
        <Field
          component={TextField}
          name="aboutMe"
          hintText="Tell us a bit about yourself: whatever you're open to sharing."
          type="text"
          multiLine={true}
          rows={5}
          rowsMax={10}
        />
        <div>
          <p>Select challenges you identify with.</p>
          {this.props.allChallenges.map((challenge, index) => this.renderChip(challenge, index))}
        </div>
        <RaisedButton label="Submit Profile" onClick={this.submitProfile} />
        <RaisedButton label="Clear Profile" onClick={this.props.clearProfile} />
      </form>
    </div>);
  }
}

const CreateProfileForm = reduxForm({
  form: 'CreateProfile',
  validate
})(CreateProfile);

export default CreateProfileForm;

CreateProfile.propTypes = {
  currentUser: PropTypes.object,
  getAllChallenges: PropTypes.func,
  allChallenges: PropTypes.array,
  form: PropTypes.object,
  addUserChallenge: PropTypes.func,
  userChallenges: PropTypes.array,
  submitProfile: PropTypes.func,
  clearProfile: PropTypes.func
};
