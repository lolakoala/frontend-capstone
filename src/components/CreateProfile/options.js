// I pulled this entire file from the redux-upload library docs.
// most things are commented out b/c they were throwing errors
// some comments are my actually commentary which will hopefully prove
// useful when we go back and look at this later.

const options = {
  // baseUrl will need to be changed when we deploy...
  // maybe handle with a process.env?
  baseUrl: 'http://localhost:3000/upload',
  // query: {
  //   category: '1',
  //   _: Date().getTime()
  // },
  query: files => {
    const length = files.length;
    const queryObj = {};
    for (let index = length - 1; index >= 0; --index) {
      queryObj[index] = files[index].name;
    }
    return queryObj;
  },
  body: {
    purpose: 'save'
  },
  // body: files => {
  //   const length = files.length;
  //   const queryObj = {};
  //   for (let index = length - 1; index >= 0; --index) {
  //     queryObj[index] = files[index].name;
  //   }
  //   return queryObj;
  // },
  dataType: 'json',
  multiple: true,
  numberLimit: 9,
  accept: 'image/*',
  // fileFieldName: 'file',
  fileFieldName: (file) => {
    return file.name;
  },
  withCredentials: false,
  requestHeaders: {
    'User-Agent': 'Warrior!'
  },
  // beforeChoose: user => {
  //   return user.isAllowUpload;
  // },
  didChoose: (files) => {
    // append this to DOM to show user they chose a file
    console.log('you choose', typeof files === 'string' ? files : files[0].name);
  },
  beforeUpload: (files) => {
    if (typeof files === 'string') return true;
    if (files[0].size < 1024 * 1024 * 20) {
      // files[0].mill = mill;
      return true;
    }
    return false;
  },
  didUpload: (files) => {
    // send file to server?
    console.log('you just uploaded', typeof files === 'string' ? files : files[0].name);
  },
  uploading: (progress) => {
    console.log('loading...', progress.loaded / progress.total + '%');
  },
  uploadSuccess: (resp) => {
    console.log('upload success!');
  },
  uploadError: (error) => {
    console.log({ error });
  }
};

export default options;
