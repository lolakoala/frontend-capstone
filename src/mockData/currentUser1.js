const currentUser1 = {
  email: 'lola@gmail.com',
  city: 'Denver',
  userName: 'lolakoala',
  aboutMe: 'Lola loves doggos.'
};

export default currentUser1;
