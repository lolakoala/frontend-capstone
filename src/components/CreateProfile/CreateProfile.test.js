import React from "react";
import CreateProfile from './CreateProfile.js';
import { shallow, configure } from "enzyme";
import currentUser1 from '../../mockData/currentUser1.js';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('CreateProfile', () => {
  const mockFn = jest.fn();
  const allChallenges = ['depression', 'anxiety'];
  const wrapper = shallow(<CreateProfile
    currentUser={{
      email: 'lola@lola.com',
      city: 'Denver'
    }}
    getAllChallenges={mockFn}
    allChallenges={allChallenges}
    addUserChallenge={mockFn}
    removeUserChallenge={mockFn}
    userChallenges={['anxiety']}
    submitProfile={mockFn}
  />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should call function on mount', () => {
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 2 buttons', () => {
    const button = wrapper.find('RaisedButton');

    expect(button.length).toEqual(2);
  });

  it('should have 2 text fields', () => {
    const textField = wrapper.find('TextField');

    expect(textField.length).toEqual(2);
  });

  it('should 1 instruction buttons', () => {
    const instruction = wrapper.find('p');

    expect(instruction.length).toEqual(1);
  });

  it('should have as many chips as challenges', () => {
    const chips = wrapper.find('Chip');

    expect(chips.length).toEqual(allChallenges.length);
  });

  it('should run function on click of challenge', () => {
    const chip = wrapper.find('Chip').first();

    chip.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(2);
  });

  it('should change state on change of inputs', () => {
    expect(wrapper.state()).toEqual({
      userName: '',
      aboutMe: ''
    });

    const firstInput = wrapper.find('TextField').first();
    const secondInput = wrapper.find('TextField').last();

    firstInput.simulate('change', { target: { value: 'lk' } });
    secondInput.simulate('change', { target: { value: 'about me' } });
    expect(wrapper.state().userName).toEqual('lk');
    expect(wrapper.state().aboutMe).toEqual('about me');
  });

  it('should fun function on submit', () => {
    const submit = wrapper.find('RaisedButton').last();

    submit.simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(3);
  });

});
