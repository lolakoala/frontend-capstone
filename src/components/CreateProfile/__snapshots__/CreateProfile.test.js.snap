// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfile should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateProfile
    addUserChallenge={[Function]}
    allChallenges={
        Array [
            "depression",
            "anxiety",
          ]
    }
    currentUser={
        Object {
            "city": "Denver",
            "email": "lola@lola.com",
          }
    }
    getAllChallenges={[Function]}
    removeUserChallenge={[Function]}
    submitProfile={[Function]}
    userChallenges={
        Array [
            "anxiety",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="pic-name-city"
>
          <div
                    className="pic-button"
          >
                    <RaisedButton
                              disabled={false}
                              fullWidth={false}
                              label="Upload Photo"
                              labelPosition="after"
                              onClick={[Function]}
                              primary={false}
                              secondary={false}
                    />
          </div>
          <div
                    className="name-city"
          >
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              fullWidth={false}
                              hintText="Username: Must be at least 5 characters."
                              multiLine={false}
                              name="userName"
                              onChange={[Function]}
                              rows={1}
                              type="text"
                              underlineShow={true}
                              value=""
                    />
                    <TextField
                              disabled={false}
                              floatingLabelFixed={false}
                              fullWidth={false}
                              hintText="Tell us a bit about yourself: whatever you're open to sharing."
                              multiLine={true}
                              name="aboutMe"
                              onChange={[Function]}
                              rows={1}
                              rowsMax={10}
                              type="text"
                              underlineShow={true}
                              value=""
                    />
          </div>
</div>,
        <div
          className="profile-bottom"
>
          <div>
                    <p>
                              Select challenges you identify with.
                    </p>
                    <Chip
                              backgroundColor=""
                              containerElement="div"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyboardFocus={[Function]}
                              onMouseDown={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              style={
                                        Object {
                                                  "marginBottom": "10px",
                                                }
                              }
                    >
                              depression
                    </Chip>
                    <Chip
                              backgroundColor="#2F9C95"
                              containerElement="div"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyboardFocus={[Function]}
                              onMouseDown={[Function]}
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchStart={[Function]}
                              style={
                                        Object {
                                                  "marginBottom": "10px",
                                                }
                              }
                    >
                              anxiety
                    </Chip>
          </div>
          <RaisedButton
                    disabled={false}
                    fullWidth={false}
                    label="Submit Profile"
                    labelPosition="after"
                    onClick={[Function]}
                    primary={false}
                    secondary={false}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="pic-button"
>
              <RaisedButton
                            disabled={false}
                            fullWidth={false}
                            label="Upload Photo"
                            labelPosition="after"
                            onClick={[Function]}
                            primary={false}
                            secondary={false}
              />
</div>,
            <div
              className="name-city"
>
              <TextField
                            disabled={false}
                            floatingLabelFixed={false}
                            fullWidth={false}
                            hintText="Username: Must be at least 5 characters."
                            multiLine={false}
                            name="userName"
                            onChange={[Function]}
                            rows={1}
                            type="text"
                            underlineShow={true}
                            value=""
              />
              <TextField
                            disabled={false}
                            floatingLabelFixed={false}
                            fullWidth={false}
                            hintText="Tell us a bit about yourself: whatever you're open to sharing."
                            multiLine={true}
                            name="aboutMe"
                            onChange={[Function]}
                            rows={1}
                            rowsMax={10}
                            type="text"
                            underlineShow={true}
                            value=""
              />
</div>,
          ],
          "className": "pic-name-city",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <RaisedButton
                disabled={false}
                fullWidth={false}
                label="Upload Photo"
                labelPosition="after"
                onClick={[Function]}
                primary={false}
                secondary={false}
/>,
              "className": "pic-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "fullWidth": false,
                "label": "Upload Photo",
                "labelPosition": "after",
                "onClick": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TextField
                  disabled={false}
                  floatingLabelFixed={false}
                  fullWidth={false}
                  hintText="Username: Must be at least 5 characters."
                  multiLine={false}
                  name="userName"
                  onChange={[Function]}
                  rows={1}
                  type="text"
                  underlineShow={true}
                  value=""
/>,
                <TextField
                  disabled={false}
                  floatingLabelFixed={false}
                  fullWidth={false}
                  hintText="Tell us a bit about yourself: whatever you're open to sharing."
                  multiLine={true}
                  name="aboutMe"
                  onChange={[Function]}
                  rows={1}
                  rowsMax={10}
                  type="text"
                  underlineShow={true}
                  value=""
/>,
              ],
              "className": "name-city",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "floatingLabelFixed": false,
                  "fullWidth": false,
                  "hintText": "Username: Must be at least 5 characters.",
                  "multiLine": false,
                  "name": "userName",
                  "onChange": [Function],
                  "rows": 1,
                  "type": "text",
                  "underlineShow": true,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "floatingLabelFixed": false,
                  "fullWidth": false,
                  "hintText": "Tell us a bit about yourself: whatever you're open to sharing.",
                  "multiLine": true,
                  "name": "aboutMe",
                  "onChange": [Function],
                  "rows": 1,
                  "rowsMax": 10,
                  "type": "text",
                  "underlineShow": true,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <p>
                            Select challenges you identify with.
              </p>
              <Chip
                            backgroundColor=""
                            containerElement="div"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyboardFocus={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                            style={
                                          Object {
                                                        "marginBottom": "10px",
                                                      }
                            }
              >
                            depression
              </Chip>
              <Chip
                            backgroundColor="#2F9C95"
                            containerElement="div"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyboardFocus={[Function]}
                            onMouseDown={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchStart={[Function]}
                            style={
                                          Object {
                                                        "marginBottom": "10px",
                                                      }
                            }
              >
                            anxiety
              </Chip>
</div>,
            <RaisedButton
              disabled={false}
              fullWidth={false}
              label="Submit Profile"
              labelPosition="after"
              onClick={[Function]}
              primary={false}
              secondary={false}
/>,
          ],
          "className": "profile-bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Select challenges you identify with.
</p>,
                Array [
                  <Chip
                    backgroundColor=""
                    containerElement="div"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyboardFocus={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                                        Object {
                                                            "marginBottom": "10px",
                                                          }
                    }
>
                    depression
</Chip>,
                  <Chip
                    backgroundColor="#2F9C95"
                    containerElement="div"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyboardFocus={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    style={
                                        Object {
                                                            "marginBottom": "10px",
                                                          }
                    }
>
                    anxiety
</Chip>,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select challenges you identify with.",
                },
                "ref": null,
                "rendered": "Select challenges you identify with.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": "depression0",
                "nodeType": "class",
                "props": Object {
                  "backgroundColor": "",
                  "children": "depression",
                  "containerElement": "div",
                  "onBlur": [Function],
                  "onClick": [Function],
                  "onFocus": [Function],
                  "onKeyDown": [Function],
                  "onKeyboardFocus": [Function],
                  "onMouseDown": [Function],
                  "onMouseEnter": [Function],
                  "onMouseLeave": [Function],
                  "onMouseUp": [Function],
                  "onTouchEnd": [Function],
                  "onTouchStart": [Function],
                  "style": Object {
                    "marginBottom": "10px",
                  },
                },
                "ref": null,
                "rendered": "depression",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "anxiety1",
                "nodeType": "class",
                "props": Object {
                  "backgroundColor": "#2F9C95",
                  "children": "anxiety",
                  "containerElement": "div",
                  "onBlur": [Function],
                  "onClick": [Function],
                  "onFocus": [Function],
                  "onKeyDown": [Function],
                  "onKeyboardFocus": [Function],
                  "onMouseDown": [Function],
                  "onMouseEnter": [Function],
                  "onMouseLeave": [Function],
                  "onMouseUp": [Function],
                  "onTouchEnd": [Function],
                  "onTouchStart": [Function],
                  "style": Object {
                    "marginBottom": "10px",
                  },
                },
                "ref": null,
                "rendered": "anxiety",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "fullWidth": false,
              "label": "Submit Profile",
              "labelPosition": "after",
              "onClick": [Function],
              "primary": false,
              "secondary": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="pic-name-city"
>
            <div
                        className="pic-button"
            >
                        <RaisedButton
                                    disabled={false}
                                    fullWidth={false}
                                    label="Upload Photo"
                                    labelPosition="after"
                                    onClick={[Function]}
                                    primary={false}
                                    secondary={false}
                        />
            </div>
            <div
                        className="name-city"
            >
                        <TextField
                                    disabled={false}
                                    floatingLabelFixed={false}
                                    fullWidth={false}
                                    hintText="Username: Must be at least 5 characters."
                                    multiLine={false}
                                    name="userName"
                                    onChange={[Function]}
                                    rows={1}
                                    type="text"
                                    underlineShow={true}
                                    value=""
                        />
                        <TextField
                                    disabled={false}
                                    floatingLabelFixed={false}
                                    fullWidth={false}
                                    hintText="Tell us a bit about yourself: whatever you're open to sharing."
                                    multiLine={true}
                                    name="aboutMe"
                                    onChange={[Function]}
                                    rows={1}
                                    rowsMax={10}
                                    type="text"
                                    underlineShow={true}
                                    value=""
                        />
            </div>
</div>,
          <div
            className="profile-bottom"
>
            <div>
                        <p>
                                    Select challenges you identify with.
                        </p>
                        <Chip
                                    backgroundColor=""
                                    containerElement="div"
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyboardFocus={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchStart={[Function]}
                                    style={
                                                Object {
                                                            "marginBottom": "10px",
                                                          }
                                    }
                        >
                                    depression
                        </Chip>
                        <Chip
                                    backgroundColor="#2F9C95"
                                    containerElement="div"
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyboardFocus={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchStart={[Function]}
                                    style={
                                                Object {
                                                            "marginBottom": "10px",
                                                          }
                                    }
                        >
                                    anxiety
                        </Chip>
            </div>
            <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Submit Profile"
                        labelPosition="after"
                        onClick={[Function]}
                        primary={false}
                        secondary={false}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="pic-button"
>
                <RaisedButton
                                disabled={false}
                                fullWidth={false}
                                label="Upload Photo"
                                labelPosition="after"
                                onClick={[Function]}
                                primary={false}
                                secondary={false}
                />
</div>,
              <div
                className="name-city"
>
                <TextField
                                disabled={false}
                                floatingLabelFixed={false}
                                fullWidth={false}
                                hintText="Username: Must be at least 5 characters."
                                multiLine={false}
                                name="userName"
                                onChange={[Function]}
                                rows={1}
                                type="text"
                                underlineShow={true}
                                value=""
                />
                <TextField
                                disabled={false}
                                floatingLabelFixed={false}
                                fullWidth={false}
                                hintText="Tell us a bit about yourself: whatever you're open to sharing."
                                multiLine={true}
                                name="aboutMe"
                                onChange={[Function]}
                                rows={1}
                                rowsMax={10}
                                type="text"
                                underlineShow={true}
                                value=""
                />
</div>,
            ],
            "className": "pic-name-city",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <RaisedButton
                  disabled={false}
                  fullWidth={false}
                  label="Upload Photo"
                  labelPosition="after"
                  onClick={[Function]}
                  primary={false}
                  secondary={false}
/>,
                "className": "pic-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "fullWidth": false,
                  "label": "Upload Photo",
                  "labelPosition": "after",
                  "onClick": [Function],
                  "primary": false,
                  "secondary": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Username: Must be at least 5 characters."
                    multiLine={false}
                    name="userName"
                    onChange={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={true}
                    value=""
/>,
                  <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Tell us a bit about yourself: whatever you're open to sharing."
                    multiLine={true}
                    name="aboutMe"
                    onChange={[Function]}
                    rows={1}
                    rowsMax={10}
                    type="text"
                    underlineShow={true}
                    value=""
/>,
                ],
                "className": "name-city",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "disabled": false,
                    "floatingLabelFixed": false,
                    "fullWidth": false,
                    "hintText": "Username: Must be at least 5 characters.",
                    "multiLine": false,
                    "name": "userName",
                    "onChange": [Function],
                    "rows": 1,
                    "type": "text",
                    "underlineShow": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "disabled": false,
                    "floatingLabelFixed": false,
                    "fullWidth": false,
                    "hintText": "Tell us a bit about yourself: whatever you're open to sharing.",
                    "multiLine": true,
                    "name": "aboutMe",
                    "onChange": [Function],
                    "rows": 1,
                    "rowsMax": 10,
                    "type": "text",
                    "underlineShow": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <p>
                                Select challenges you identify with.
                </p>
                <Chip
                                backgroundColor=""
                                containerElement="div"
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyboardFocus={[Function]}
                                onMouseDown={[Function]}
                                onMouseEnter={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchStart={[Function]}
                                style={
                                                Object {
                                                                "marginBottom": "10px",
                                                              }
                                }
                >
                                depression
                </Chip>
                <Chip
                                backgroundColor="#2F9C95"
                                containerElement="div"
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onKeyDown={[Function]}
                                onKeyboardFocus={[Function]}
                                onMouseDown={[Function]}
                                onMouseEnter={[Function]}
                                onMouseLeave={[Function]}
                                onMouseUp={[Function]}
                                onTouchEnd={[Function]}
                                onTouchStart={[Function]}
                                style={
                                                Object {
                                                                "marginBottom": "10px",
                                                              }
                                }
                >
                                anxiety
                </Chip>
</div>,
              <RaisedButton
                disabled={false}
                fullWidth={false}
                label="Submit Profile"
                labelPosition="after"
                onClick={[Function]}
                primary={false}
                secondary={false}
/>,
            ],
            "className": "profile-bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Select challenges you identify with.
</p>,
                  Array [
                    <Chip
                      backgroundColor=""
                      containerElement="div"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyboardFocus={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      style={
                                            Object {
                                                                  "marginBottom": "10px",
                                                                }
                      }
>
                      depression
</Chip>,
                    <Chip
                      backgroundColor="#2F9C95"
                      containerElement="div"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyboardFocus={[Function]}
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      style={
                                            Object {
                                                                  "marginBottom": "10px",
                                                                }
                      }
>
                      anxiety
</Chip>,
                  ],
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Select challenges you identify with.",
                  },
                  "ref": null,
                  "rendered": "Select challenges you identify with.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": "depression0",
                  "nodeType": "class",
                  "props": Object {
                    "backgroundColor": "",
                    "children": "depression",
                    "containerElement": "div",
                    "onBlur": [Function],
                    "onClick": [Function],
                    "onFocus": [Function],
                    "onKeyDown": [Function],
                    "onKeyboardFocus": [Function],
                    "onMouseDown": [Function],
                    "onMouseEnter": [Function],
                    "onMouseLeave": [Function],
                    "onMouseUp": [Function],
                    "onTouchEnd": [Function],
                    "onTouchStart": [Function],
                    "style": Object {
                      "marginBottom": "10px",
                    },
                  },
                  "ref": null,
                  "rendered": "depression",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "anxiety1",
                  "nodeType": "class",
                  "props": Object {
                    "backgroundColor": "#2F9C95",
                    "children": "anxiety",
                    "containerElement": "div",
                    "onBlur": [Function],
                    "onClick": [Function],
                    "onFocus": [Function],
                    "onKeyDown": [Function],
                    "onKeyboardFocus": [Function],
                    "onMouseDown": [Function],
                    "onMouseEnter": [Function],
                    "onMouseLeave": [Function],
                    "onMouseUp": [Function],
                    "onTouchEnd": [Function],
                    "onTouchStart": [Function],
                    "style": Object {
                      "marginBottom": "10px",
                    },
                  },
                  "ref": null,
                  "rendered": "anxiety",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "fullWidth": false,
                "label": "Submit Profile",
                "labelPosition": "after",
                "onClick": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
